code/src/containers/forms/product-form/hooks/use-update-product/useUpdateProduct.test.ts

```diff
@@ -0,0 +1,86 @@
+import { renderHook } from "@testing-library/react";
+
+import useUpdateProduct from "@/containers/forms/product-form/hooks/use-update-product/useUpdateProduct";
+
+import routes from "@/constants/routes";
+
+const mockUnwrap = jest.fn();
+const mockNavigate = jest.fn();
+const mockOpenSnackbarWithTimeout = jest.fn();
+const mockUpdateProduct = jest.fn(() => ({ unwrap: mockUnwrap }));
+
+jest.mock("react-router-dom", () => ({
+  useNavigate: () => mockNavigate
+}));
+
+jest.mock("@/hooks/use-snackbar/useSnackbar", () => ({
+  __esModule: true,
+  default: () => ({
+    openSnackbarWithTimeout: mockOpenSnackbarWithTimeout
+  })
+}));
+
+jest.mock("@/store/api/productsApi", () => ({
+  useUpdateProductMutation: () => [mockUpdateProduct, {}]
+}));
+
+const successSnackbarConfig = {
+  variant: "success",
+  messageTranslationKey: "productForm.updation.success"
+};
+
+const errorSnackbarConfig = {
+  variant: "error",
+  messageTranslationKey: "productForm.updation.fail"
+};
+
+describe("Test useUpdateProduct hook", () => {
+  test("Should return correct values", () => {
+    const { result } = renderHook(() => useUpdateProduct());
+
+    const [onSubmit, requestState] = result.current;
+
+    expect(onSubmit).toBeTruthy();
+    expect(requestState).toBeTruthy();
+  });
+
+  test("Should call update function with right parameters", () => {
+    const { result } = renderHook(() => useUpdateProduct());
+
+    const [onSubmit] = result.current;
+
+    onSubmit({ productId: "1" });
+
+    expect(mockUpdateProduct).toHaveBeenCalledWith({ productId: "1" });
```

Comment:
Object is repeated. Create variable and reuse

Reply:
Thank you, fixed